DCOS/fluentd-testing/fluentd.Tests.ps1

```diff
@@ -0,0 +1,61 @@
+﻿
+$here = Split-Path -Parent $MyInvocation.MyCommand.Path
+$newConfigPath = Join-Path $here td-agent.conf
+
+function Get-ServiceConfPath() {
+    $serviceInfo = Get-ItemProperty -PATH HKLM:\SYSTEM\CurrentControlSet\Services\fluentdwinsvc
+    $confPath = $serviceInfo.fluentdopt -split " " | where {$_ -like "*.conf"}
+    return $confPath
+}
+
+Describe "Fluentd sanity check" {
+
+    It "Service can be found" {
+        $service = Get-Service "fluentdwinsvc"
+        $service | Should -Not -Be $null
+    }
+
+    It "Service is running" {
+        $service = Get-Service "fluentdwinsvc"
+        $service.Status | Should -Be "Running"
+    }
+
+    It "Can find custom config in current directory" {
+        Test-Path $newConfigPath -PathType Leaf | Should -Be $true
+    }
+
+    It "Can get get service opts" {
+        Get-ServiceConfPath | Should -Not -Be $null
+    }
+}
+
+
+Describe "Fluentd logging" {
+
+    It "Can modify conf file" {
+        Copy-Item $newConfigPath $(Get-ServiceConfPath) -force
+        $LASTEXITCODE | Should -Be 0
```

Comment:
`Copy-Item` from above is a PowerShell cmdlet and it won't set the `$LASTEXITCODE` as it throws exceptions on errors. So this check can be safely removed.

I would say the correct check is:

```
{ Copy-Item $newConfigPath $(Get-ServiceConfPath) -force } | Should Not Throw
```

This is an [example](https://github.com/cloudbase/juju-powershell-modules/blob/9a5af4f1e1be182daae2739ef2fc090fa274a3f6/JujuHelper/Tests/JujuHelper.Tests.ps1#L41) of an assertion of this type.